### ISSUE TITLE
--csv not too useful on files with different lines

### ISSUE THREAD

**Issue by chirlu** [rustyrussell/stats#4]
The `--csv` option is currently not very useful when applied to a file with several different lines, e.g.:

```
$ head -n 7 test/test.in | stats --csv
" [1] mon"
"this [2] day"
" [3] [4]"
100
150
-100
200
300
100,200
200,300
```

There is no way to find out where the values for `[1] mon` end and those for `this [2] day` begin. The values for the third line can be told apart, but only accidentally.

Using `--count` (see #3) does mitigate the issue somewhat, but it’s still less than ideal:

```
$ head -n 7 test/test.in | stats --csv -c
" [1] mon"  (2)
"this [2] day"  (3)
" [3] [4]"  (2)
100
(...)
```

My ideas how to solve this:
1. Just give a warning.
2. Reorder the output so that it can be split easily, possibly with a blank line (but “a line starting with `"`” would also be unambiguous):

```
" [1] mon"
100
150

"this [2] day"
-100
200
300

" [3] [4]"
100,200
200,300
```

Thoughts? I could implement either myself, of course.

**Comment by rustyrussell** [comment]
Yeah, I've already changed how --csv works once, and I'm still not happy with it.  I think your latter solution looks like a really neat option!

I'd like to see the patch :)

Cheers,
Rusty.

**Comment by chirlu** [comment]
Now in #7. The output format change is actually just 8ccd16ab636ac695f59f514e21eeda8cb371bbca, but then I got carried away a bit …

**Comment by rustyrussell** [comment]
BTW, if you'd put 'Fixes #4' in the commit message, it would have automatically closed this issue, too!
